design/volume-group-snapshot.md

```diff
@@ -0,0 +1,428 @@
+    - At first it checks whether the PVC is bound and VolumeName is non-empty.
+    - Then it adds the related PV under the list of relatedItems.
+    - Following on, the plugin adds the pods mounting the PVC as relatedItems.
+    - Now we need to extend this PVC IBA plugin to add the PVCs to be grouped for a particular VGS object, so that they are processed together under an ItemBlock by Velero.
+        - First we will check if the PVC that is being processed by the plugin has the user specified VGS label.
+        - If it is present then we will execute a List call in the namespace with the label as a matching criteria and see if this results in any PVCs (other than the current one).
+        - If there are PVCs matching the criteria then we add the PVCs to the relatedItems list.
+        - This helps in building the ItemBlock we need for VGS processing, i.e. we have the relevant pods and PVCs in the ItemBlock.
+
+- Updates to CSI PVC plugin:
+    - This is the plugin which creates VolumeSnapshots for the CSI PVCs if required.
+    - We will have to update this plugin to:
+        - Create VGS for the group of PVCs
+            - At First we will check if PVC has the user specified VGS label.
+            - If yes, then it implies it needs to be part of a VGS instance.
+            - Check if VGS exists for this PVC or not, we don't want to recreate VGS object.
+            - We can check if VGS exists by listing the VGS in the namespace using the backup UUID alongwith the label used for the VGS (we will be creating VGS with backup UUID and user specified VGS  group label) and see if the PVC is part of any of the existing VGS for the particular backup operation we are currently in.
+            - If not then create the VGS object with the required labels (in this case backup UUID and the user specified VGS group label) and also add the PVC names (that are included in the VGS) as annotations. (This will be useful for our VGS restore operation where we would need PVC resource identifiers).
+            - Once VGS is created, add the VGS as an additional item to be included in the backup.
+        - Skip creation of VS for PVC that are already included in VGS
+            - Now if the PVC has the user specified VGS label that means it is destined to be part of some VGS and the VS would be created by the K8s VGS controller and Velero does not need to explicitly create the VS for the PVC.
+
+- Add a new VolumeGroupSnapshot(VGS) BIA plugin
+    - For a particular instance of VGS, this plugin will wait for all the related VS instances to be in ready state.
+    - And once they are ready, add all the VS instances as additional items for the backup.
+    - This plugin will also add VolumeGroupSnapshotClass(VGSClass) as an additional item to be included in the backup.
+
+![VGS-Backup-Sequence](VGS-Backup-Sequence.png)
+
+
+Restore workflow:
```

Comment:
As far as I understand, this is not required by data mover.
Data mover's Generic Restore process is neutral to how the backup goes.

Let's clarify this.